src/templates/seo-head.js

```diff
@@ -21,71 +21,127 @@ export const nuxtI18nSeo = function () {
     return {}
   }
   // Prepare html lang attribute
-  const currentLocaleData = this.$i18n.locales.find(l => l[LOCALE_CODE_KEY] === this.$i18n.locale)
+  const currentLocale = this.$i18n.locales.find(l => codeFromLocale(l) === this.$i18n.locale)
+  const currentLocaleIso = isoFromLocale(currentLocale)
+
   const htmlAttrs = {}
-  if (currentLocaleData && currentLocaleData[LOCALE_ISO_KEY]) {
-    htmlAttrs.lang = currentLocaleData[LOCALE_ISO_KEY]
+
+  if (currentLocale && currentLocaleIso) {
+    htmlAttrs.lang = currentLocaleIso // TODO: simple lang or "specific" lang with territory?
   }
 
   const link = []
-  // hreflang tags
-  if (strategy !== STRATEGIES.NO_PREFIX) {
-    link.push(...this.$i18n.locales
-      .map(locale => {
-        if (locale[LOCALE_ISO_KEY]) {
-          return {
-            hid: 'alternate-hreflang-' + locale[LOCALE_ISO_KEY],
-            rel: 'alternate',
-            href: baseUrl + this.switchLocalePath(locale.code),
-            hreflang: locale[LOCALE_ISO_KEY]
-          }
-        } else {
-          // eslint-disable-next-line no-console
-          console.warn(`[${MODULE_NAME}] Locale ISO code is required to generate alternate link`)
-          return null
-        }
-      })
-      .filter(item => !!item))
-  }
 
-  // canonical links
-  if (strategy === STRATEGIES.PREFIX_AND_DEFAULT) {
-    const canonicalPath = this.switchLocalePath(currentLocaleData[LOCALE_CODE_KEY])
-    if (canonicalPath && canonicalPath !== this.$route.path) {
-      // Current page is not the canonical one -- add a canonical link
-      link.push({
-        hid: 'canonical-lang-' + currentLocaleData[LOCALE_CODE_KEY],
-        rel: 'canonical',
-        href: baseUrl + canonicalPath
-      })
-    }
-  }
+  addHreflangLinks.bind(this)(currentLocale, this.$i18n.locales, link)
```

Comment:
I tried to restructure the code and went with functions + explicit binding instead of arrow functions.

I've also thought about returning a part of the `link`/`meta` array and eventually concatenating them instead of altering/pushing directly

Would love to get your feedback on that. ☺️

Reply:
It looks fine. :)